pallets/accelerated-bn-crypto/src/tests.rs

```diff
@@ -0,0 +1,121 @@
+// Copyright 2024, Horizen Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! Generic executions of the operations for *Arkworks* elliptic curves.
+
+// As not all functions are used by each elliptic curve and some elliptic
+// curve may be excluded by the build we resort to `#[allow(unused)]` to
+// suppress the expected warning.
+
+use crate::{
+    mock::*,
+    utils::{make_msm_args, make_pairing_args, make_scalar_args, make_scalar_args_projective},
+};
+use codec::Encode;
+use frame_support::assert_ok;
+
+const SCALAR_WORDS: u32 = 3;
+const MSM_LEN: u32 = 10;
+
+// ---------------------------------------------
+// Tests for BN254
+// ---------------------------------------------
+
```

Comment:
Instead of using a comment, I think it's more idiomatic to enclose all these tests inside a `mod bn254`, and remove the `bn254_` prefix from all the test functions.